slave/components/quality_controller/error_correction.go

```diff
@@ -21,16 +32,100 @@ type errorCorrection struct {
 }
 
 // NewErrorCorrection constructs and returns an errorCorrection struct, which implements the Controller interface
-func NewErrorCorrection(ctx context.Context, dockerCli *client.Client, config *config_parser.Config, fileToDecontaminate string) Controller {
+func NewErrorCorrection(ctx context.Context, dockerCli *client.Client, config *config_parser.Config, fileToCorrect string) Controller {
 	return &errorCorrection{
 		ctx:             ctx,
 		dockerCLI:       dockerCli,
 		config:          config,
-		toDecontaminate: fileToDecontaminate,
+		toDecontaminate: fileToCorrect,
+	}
+}
+
+// TODO: Replace this function into a constants file where this can be used by other quality control processes
+// getImageID attempts to find the Docker image by given term
+func getImageID(client *client.Client, ctx context.Context, term string) (string, error) {
+	images, err := client.ImageList(ctx, types.ImageListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to get available Docker images, err: %v", err)
+	} else if len(images) == 0 {
+		return "", fmt.Errorf("getImageID found no images")
+	}
+	found := false
+	assemblerID := ""
+	for _, im := range images {
+		if found {
+			break
+		}
+		for _, tag := range im.RepoTags {
+			if strings.Contains(tag, term) {
+				found = true
+				assemblerID = im.ID
+			}
+		}
 	}
+	if !found {
+		return "", fmt.Errorf("failed to find a Docker container for the given assembler")
+	}
+	return assemblerID, nil
 }
 
 // Process performs the error correction process
 func (e *errorCorrection) Process() (string, error) {
-	return "", nil
+	// correctedFile is the placeholder filename where the corrected reads are going to be stored.
+	var correctedFile = path.Join(constants.BaseOut, "canu-corr", "correction", "run1.corrected.fastq")
```

Comment:
This is going to be `/output/canu-corr/correction/run1.corrected.fastq`. Is it necessary? Why not `/output/corrected.fastq`, similar to `/output/trimmed.fastq`?

Reply:
Canu has a specific pattern of saving output files which cannot be changed.